Template activation: improve UX for custom templates

mcsf opened the pull request:
## What?

Follow-up of #67125

In the template editor:

* Disable the "Activate / Deactivate" action for custom templates
* Don't try to label custom templates' status as "Active" or "Inactive"; use "Custom" instead

## Why?

Custom templates aren't part of the template hierarchy and thus aren't subject to activation. The actions available and UI should reflect this.

## Questions

- [ ] In the Status field, I chose to use a Badge component with `intent="info"` to set custom templates apart from deactivated templates. What do you think of this? Happy to remove it and let the default take over.

## Screenshots or screencast 

|Before|After|
|-|-|
|<img width="1777" height="652" alt="Screenshot 2025-10-20 at 16 56 05" src="https://github.com/user-attachments/assets/7ff16996-5ec9-4571-b0f3-d8e330458083" />|<img width="1777" height="652" alt="Screenshot 2025-10-20 at 16 55 45" src="https://github.com/user-attachments/assets/9922b4b4-b3bb-4268-9e55-56c9b05df938" />|
|<img width="294" height="449" alt="Screenshot 2025-10-20 at 16 56 33" src="https://github.com/user-attachments/assets/5820421b-daeb-439d-a93a-71c9d4dc6e22" />|<img width="294" height="449" alt="Screenshot 2025-10-20 at 16 56 48" src="https://github.com/user-attachments/assets/be318bf7-f8f1-4dba-9e04-d2b47db0c11e" />|

mcsf committed:
Template activation: improve UX for custom templates

juanfra commented on packages/edit-site/src/components/page-templates/fields.js:
```diff
@@ -155,6 +155,11 @@ export const activeField = {
+		if ( item.is_custom ) {
+			// translators: Refers to a type of template: custom template
+			return <Badge intent="info">{ __( 'Custom' ) }</Badge>;
+		}
```
As a site admin, I might not fully understand what “Custom” means when it appears next to the status. I’d associate “Custom” with a template type instead, which is also pixels away from the status, and could feel redundant.

My expectation would be to see the actual status (active or inactive) and, if the template type is custom, get some context on why I can or can’t make changes.

Maybe a padlock icon on the active/inactive status for custom templates, with a tooltip explaining the limitation, could work well here.

Also, labeling a status as “Custom” might suggest it’s something I can create or edit, which could be confusing.

mcsf replied:
Thanks for the feedback, @juanfra!

My initial version had a more basic "N/A" label instead. Because custom templates really don't have a status. At least right now they don't; maybe at some point we could attempt to track them to posts that use them, but none of that is built.

> padlock icon

From a design / components standpoint, I don't know if we can do this, I'd like input from designers.

> with a tooltip explaining the limitation, could work well here

This makes perfect sense, I think we should. However, I'm working with @ellatrix and we'd like to merge something imperfect _now_ and polish those interactions during the Beta cycle.

I'll quickly address the `_x()` part, but I hope you can approve this more or less as-is. Yay or nay on the "N/A" label?

juanfra replied:
Thanks for the response, Miguel! I understand the urgency with 6.9 around the corner, so I agree that “N/A” works better than the “Custom” approach. I’m good approving this as-is, with the hope that we can revisit and refine it in the future.

mcsf replied:
Much appreciated! :)

juanfra commented on packages/edit-site/src/components/page-templates/fields.js:
```diff
@@ -192,7 +197,7 @@ export const slugField = {
 		return (
 			defaultTemplateType?.title ||
-			// translators: %s is the slug of a custom template.
+			// translators: Refers to a type of template: custom template
```
I believe the context, if needed, should be passed using `_x()`. It looks like this might be a leftover from an earlier `sprintf()`.

juanfra reviewed (commented):
Thanks for the PR! I left a couple of comments.

mcsf committed:
Address feedback:

- Avoid "Custom" label for template status
- Add a tentative "title" attribute to the status badge for custom
  templates
- Use _x() where appropriate

mcsf committed:
Detect custom CREATED templates as well as REGISTERED ones

We'll want to expose this via REST API, most likely.

mcsf commented on packages/edit-site/src/components/page-templates/index.js:
```diff
@@ -150,8 +154,14 @@ export default function PageTemplates() {
 				( template ) => template.id === record.id
 			),
+			_isCustom:
+				record.is_custom ||
+				( ! record.meta?.is_wp_suggestion &&
+					! defaultTemplateTypes.find(
+						( type ) => type.slug === record.slug
+					) ),
```
Adapted from https://github.com/WordPress/gutenberg/blob/c1f90a606e516ca99ea973c04e18dde64f020eaf/packages/editor/src/components/post-template/hooks.js#L95-L102, per @ellatrix's tip.

ellatrix reviewed (approved):
Works, tested both created and registered templates.

ellatrix commented on packages/edit-site/src/components/dataviews-actions/index.js:
```diff
@@ -35,6 +35,7 @@ export const useSetActiveTemplateAction = () => {
 			icon: pencil,
 			isEligible( item ) {
 				return (
+					! item.is_custom &&
```
Should we update this?

priethor reviewed (approved):
The [failing tests](https://github.com/WordPress/gutenberg/actions/runs/18681765812/job/53264325125?pr=72499) are unrelated to this PR and started failing in trunk earlier today, seemingly after #72487. Let's merge this without those tests passing to get ready for Beta1.

ellatrix commented:
Yes

```
Notice:   2 failed
    [chromium] › test/e2e/specs/editor/various/color-support.spec.js:11:6 › Link color in themes › should apply selected link color in editor and frontend 
    [chromium] › test/e2e/specs/editor/various/content-only-lock.spec.js:80:2 › Content-only lock › should be able to automatically stop temporarily modify as blocks when an outside block is selected 
```

These are failing on trunk too and we need this PR to get merged and included in a package asap. Let's merge. I will skip the performance test too.

tyxla commented on packages/edit-site/src/components/page-templates/fields.js:
```diff
@@ -155,6 +155,19 @@ export const activeField = {
+				<Badge
+					intent="info"
+					title={ __(
+						'Custom templates cannot be active nor inactive.'
```
Worth double-checking, but I think this is grammatically wrong, and it should be this instead:

```
Custom templates cannot be active or inactive.
```
